### ISSUE TITLE
use $elemMatch when query on elements in an array field

### ISSUE THREAD

**Issue by jannyHou** [strongloop/loopback-connector-cloudant#78]
$elemMatch: Matches and returns all documents that contain an array field with at least one element that matches all the specified query criteria.

Use case:
*seed data*
```js
{
      seq: 0,
      name: 'John Lennon',
      friends: [
        {name: 'Paul McCartney'},
        {name: 'George Harrison'},
        {name: 'Ringo Starr'},
      ],
    },
```
*wrong query*
`{where: {friends.name: 'Ringo Starr'}}`
*right query*
`{where: {friends: {$elemMatch: {name: 'Ringo Starr'}}}}`

**Comment by jannyHou** [comment]
Reopen this issue since the previous fix only works when model is created in code.

Several problem with the previous pr, which need to be fixed:

- [ ] The filter only works with apis in code but doesn't work with explorer
- [ ] The previous pr is based on an assumption that the connector can get all dataType of nested properties. If the model is created in code, like what test case does, all dataType are available, while when creating model by cli, loading it from the model.json file, for some reason deep nested property's type are undefined.
- [ ] The previous pr doesn't catch some undefined situation.

**Comment by jannyHou** [comment]
UPDATE: 
Before solving the bugs mentioned in https://github.com/strongloop/loopback-connector-cloudant/issues/78#issuecomment-284454583 we are not sure should we wrap the array property by detecting property type in loopback, or add a detailed doc to tell the user to include `elemMatch` in their filter?

cc @raymondfeng

**Comment by jannyHou** [comment]
fixed in PR#90
